File: src/test/java/com/drpicox/game/testSteps/players/UseTheBrowserOf.java
package com.drpicox.game.testSteps.players;

import com.drpicox.game.forms.NewGameForm;
import com.drpicox.game.forms.VisibleGameForm;
import com.drpicox.game.testPost.SnapshotService;
import com.drpicox.game.testPost.TestPostForms;
import com.drpicox.game.testPost.reader.PostLine;
import com.drpicox.game.testSteps.AbstractPostLineStep;
import org.springframework.stereotype.Component;

@Component
public class UseTheBrowserOf extends AbstractPostLineStep {

    private final TestPostForms testPostForms;
    private final SnapshotService snapshotService;

    public UseTheBrowserOf(TestPostForms testPostForms, SnapshotService snapshotService) {
        this.testPostForms = testPostForms;
        this.snapshotService = snapshotService;
    }

    @Override
    protected String getRegex() {
        return "Use the browser of _([^_]+)_";
    }

    @Override
    protected void run(PostLine line, String[] match) {
        var newGame = testPostForms.getForm(NewGameForm.class);
        var gameName = newGame.getGameName();
        var playerName = match[1];
        var result = snapshotService.get("/api/v1/games/"+gameName+"/players/" + playerName, null, VisibleGameForm.class);
        testPostForms.setForm(result);
    }
}

File: src/www/__test__/testSteps/players/UseTheBrowserOf.ts
import { ReduxStore } from "www/ducks";
import { getGameName, enterGame } from "www/ducks/game";
import { Injector } from "www/injector";
import { AbstractPostLineStep } from "../AbstractPostLineStep";

export class UseTheBrowserOf extends AbstractPostLineStep {
  private reduxStore: ReduxStore;

  constructor(injector: Injector) {
    super(/Use the browser of _([^_]+)_/i);

    this.reduxStore = injector.get(ReduxStore);
  }

  async runMatch(_: any, match: string[]) {
    const [, playerName] = match;

    var gameName = this.reduxStore.select(getGameName);
    this.reduxStore.dispatch(enterGame({ gameName, playerName }));
  }
}
